presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkPagesSerde.java

```diff
@@ -0,0 +1,217 @@
+            while (remainingRows > 0) {
+                int rows = 50 + random.nextInt(450); // 50 - 500 rows per pass
+                List<Object>[] testRows = generateTestRows(random, ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, VARCHAR), rows);
+                remainingRows -= rows;
+                for (int i = 0; i < testRows.length; i++) {
+                    BIGINT.writeLong(pageBuilder.getBlockBuilder(0), i);
+                    writeRow(testRows[i], pageBuilder.getBlockBuilder(1));
+                    writeRow(testRows[i], pageBuilder.getBlockBuilder(2));
+                    writeRow(testRows[i], pageBuilder.getBlockBuilder(3));
+                }
+                pageBuilder.declarePositions(rows);
+                pages.add(pageBuilder.build());
+                pageBuilder.reset();
+            }
+            return pages.toArray(Page[]::new);
+        }
+
+        private void writeRow(List<Object> testRow, BlockBuilder rowBlockBuilder)
+        {
+            BlockBuilder singleRowBlockWriter = rowBlockBuilder.beginBlockEntry();
+            for (Object fieldValue : testRow) {
+                if (fieldValue == null) {
+                    singleRowBlockWriter.appendNull();
+                }
+                else if (fieldValue instanceof String) {
+                    VARCHAR.writeSlice(singleRowBlockWriter, utf8Slice((String) fieldValue));
+                }
+                else {
+                    throw new UnsupportedOperationException();
+                }
+            }
+            rowBlockBuilder.closeEntry();
+        }
+
+        // copied & modifed from TestRowBlock
+        private List<Object>[] generateTestRows(Random random, List<Type> fieldTypes, int numRows)
+        {
+            @SuppressWarnings("unchecked")
+            List<Object>[] testRows = new List[numRows];
+            for (int i = 0; i < numRows; i++) {
+                List<Object> testRow = new ArrayList<>(fieldTypes.size());
+                for (int j = 0; j < fieldTypes.size(); j++) {
+                    if (fieldTypes.get(j) == VARCHAR) {
+                        byte[] data = new byte[random.nextInt(128)];
+                        if (data.length == 0) {
+                            testRow.add(null);
+                        }
+                        else if (data.length < 32 && i > 0) {
+                            // Repeat values to make compression more interesting
+                            testRow.add(testRows[i - 1].get(j));
+                        }
+                        else {
+                            random.nextBytes(data);
+                            testRow.add(new String(data, ISO_8859_1));
+                        }
+                    }
+                    else {
+                        throw new UnsupportedOperationException();
+                    }
+                }
+                testRows[i] = testRow;
+            }
+            return testRows;
+        }
+    }
+
```

Comment:
add a test method for benchmarks, see: `io.prestosql.orc.BenchmarkOrcDecimalReader#testReadDecimal`

Reply:
Test added.